contracts/Strategy.sol

```diff
@@ -90,17 +90,29 @@ contract Strategy is AccessControl, ERC20Rewards, StrategyMigrator { // TODO: I'
         state = target;
     }
 
+    /// @dev State and state variable management
+    /// @param target State to transition to
+    function _transition(State target) internal {
```

Comment:
💯  clever solution for this

Reply:
what about adding a final `else` to the main `_transition(State target, IPool pool_)` above to future proof it from ever getting in a bad state, that would make it a true state machine imo.

```
    } else {
          revert("Invalid state");
```

Reply:
I'm using an `enum` and it is an internal function, there is no way of passing a wrong state.